packages/features/src/web-connector/utils/render-payload.tsx

```diff
@@ -0,0 +1,319 @@
+      } else if ("value" in value) {
+        simplified[key] = value.value
+      } else {
+        simplified[key] = value
+      }
+    } else {
+      simplified[key] = value
+    }
+  }
+  return simplified
+}
+
+export const renderPayload = (payload: string) => {
+  try {
+    const originalPayload = recoverOriginalPayload(payload)
+    const parsedPayload = JSON.parse(originalPayload)
+
+    // Handle presentation request format
+    if (containsPresentationRequest(parsedPayload)) {
+      const request = parsedPayload.presentationRequest
+
+      // Construct verifierUrl if it's a zk-app request and verifierIdentity is a zkAppAccount
+      const verifierUrl =
+        request.type === "zk-app" &&
+        typeof parsedPayload.verifierIdentity === "object"
+          ? `minascan.io/${parsedPayload.verifierIdentity.network}/account/${parsedPayload.verifierIdentity.address}`
+          : undefined
+
+      const formatted = [
+        `Type: ${request.type}`,
+        "",
+        formatInputsHumanReadable(request.spec.inputs),
+        "",
+        `Requirements:\n${formatLogicNode(request.spec.logic.assert, 0)}`,
+        "",
+        `Output:\n${formatLogicNode(request.spec.logic.outputClaim, 0)}`,
+        formatClaimsHumanReadable(request.claims),
+        request.inputContext
+          ? `\nContext:\n- Type: ${request.inputContext.type}\n- Action: ${typeof request.inputContext.action === "string" ? request.inputContext.action : request.inputContext.action.value}\n- Server Nonce: ${request.inputContext.serverNonce.value}`
+          : "",
+        "origin" in parsedPayload ? `\nOrigin: ${parsedPayload.origin}` : "",
+        parsedPayload.verifierIdentity
+          ? `\nVerifier Identity: ${
+              request.type === "zk-app"
+                ? JSON.stringify(parsedPayload.verifierIdentity, null, 2)
+                : parsedPayload.verifierIdentity
+            }`
+          : "",
+        verifierUrl ? `\nSee verifier on Minascan: https://${verifierUrl}` : "",
+      ].join("\n")
+
+      return <div className="whitespace-pre-wrap break-all">{formatted}</div>
+    }
+
+    // Handle credential format
+    if (isCredential(parsedPayload)) {
+      const credentialData =
+        parsedPayload.credential.value?.data || parsedPayload.credential.data
+      const simplifiedData = simplifyCredentialData(credentialData)
+      const description = parsedPayload.metadata?.description
+
+      return (
+        <div className="space-y-4">
+          {description && <p className="text-sm">{description}</p>}
+          <pre className="whitespace-pre-wrap break-all bg-neutral-900 p-3 rounded">
+            {JSON.stringify(simplifiedData, null, 2)}
+          </pre>
+        </div>
+      )
+    }
+
+    // Default fallback
+    return <pre className="whitespace-pre-wrap break-all">{payload}</pre>
+  } catch (error) {
+    return (
+      <pre className="whitespace-pre-wrap break-all">{`Error: ${error}`}</pre>
+    )
+  }
```

Comment:
the error message isn't sanitized, couldn't that cause an XSS problem similar to why you sanitize the `payload`?